GoogleCloudPlatform/k8s-config-connector
# Feat: Adding GRPC expander interface and Jinja2 grpc expander service

### Change description

Adding GRPC expander interface and Jinja2 grpc expander service

- Define GRPC proto for expander interface
  - Add make targets
  - compile proto and generate golang bindings
- Add grpc based jinja2 expander service
  - calls out to jinja2 cli
  - error handling
  - Add unit-tests to verify expander works as intended
    - window1: make docker-run-expander-jinja2
    - window2: cd expanders/jinja2; go test -v
- Add grpc expand cli
  - For quick sanity check
- Split makefiles for simple organization
  - Add inline.mk, expanders.mk


#### Next Steps

- Add support in composition-engine to callout to grpc interface as part of processing compositions
- Add manifests to start jinja2-grpc service

### Tests

```
❯ go test -v
=== RUN   TestValidate
--- PASS: TestValidate (0.00s)
=== RUN   TestEvaluateEmptyConfig
--- PASS: TestEvaluateEmptyConfig (0.12s)
=== RUN   TestEvaluateEmptyContext
--- PASS: TestEvaluateEmptyContext (0.14s)
=== RUN   TestEvaluateEmptyFacade
--- PASS: TestEvaluateEmptyFacade (0.00s)
=== RUN   TestEvaluateEmptyValue
--- PASS: TestEvaluateEmptyValue (0.12s)
=== RUN   TestEvaluateTemplateUsesFacade
--- PASS: TestEvaluateTemplateUsesFacade (0.12s)
=== RUN   TestEvaluateTemplateMissingFacadeField
--- PASS: TestEvaluateTemplateMissingFacadeField (0.14s)
=== RUN   TestEvaluateTemplateUsesContext
--- PASS: TestEvaluateTemplateUsesContext (0.13s)
=== RUN   TestEvaluateTemplateMissingContextField
--- PASS: TestEvaluateTemplateMissingContextField (0.13s)
=== RUN   TestEvaluateTemplateUsesValues
--- PASS: TestEvaluateTemplateUsesValues (0.12s)
=== RUN   TestEvaluateTemplateMissingValuesField
--- PASS: TestEvaluateTemplateMissingValuesField (0.13s)
=== RUN   TestEvaluateTemplateWrongTopLevelField
--- PASS: TestEvaluateTemplateWrongTopLevelField (0.13s)
PASS
ok      google.com/composition/expanders/jinja2 1.320s
```
